python/lsst/afw/geom/testUtils.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import, division
+#
+# LSST Data Management System
+# Copyright 2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""Utilities that should be imported into the lsst.afw.coord namespace when lsst.afw.coord is used
+
+In the case of the assert functions, importing them makes them available in lsst.utils.tests.TestCase
+"""
+import numpy as np
+
+from .geomLib import Box2I, Box2D
+
+__all__ = ["BoxGrid"]
+
+class BoxGrid(object):
+    """!Divide a box into nx x ny smaller boxes
+    """
+    def __init__(self, box, numColRow):
+        """!Construct a BoxGrid
+
+        @param[in] box  box (an lsst.afw.geom.Box2I or Box2D);
+                        the boxes in the grid will be of the same type
+        @param[in] numColRow  number of columns and rows
+        """
+        if len(numColRow) != 2:
+            raise RuntimeError("numColRow=%r; must be a sequence of two integers" % (numColRow,))
+        self._numColRow = tuple(int(val) for val in numColRow)
+
+        boxClass = type(box)
+        if boxClass == Box2I:
+            stopDelta = 1
+        elif boxClass == Box2D:
+            stopDelta = 0
+        else:
+            raise RuntimeError("Unknown box class %s of box %s" % (boxClass, box))
+        self.boxClass = boxClass
+        self.stopDelta = stopDelta
+
+        minPoint = box.getMin()
+        self.pointClass = type(minPoint)
+        dtype = np.array(minPoint).dtype
+
+        self._divList = [np.linspace(start=box.getMin()[i],
+                                     stop=box.getMax()[i] + self.stopDelta,
+                                     num=self._numColRow[i] + 1,
+                                     endpoint=True,
+                                     dtype=dtype) for i in range(2)]
+
+    @property
+    def numColRow(self):
+        return self._numColRow
```

Comment:
To make it read only. This class looks a bit like a numpy array, which you can reshape on the fly. I wanted to avoid anyone attempting that.